datafusion/sqllogictest/test_files/nullif.slt

```diff
@@ -97,11 +97,35 @@ SELECT NULLIF(1, 3);
 ----
 1
 
-query I
+query T
 SELECT NULLIF(NULL, NULL);
 ----
 NULL
 
+query R
+select nullif(1, 1.2);
+----
+1
+
+query R
+select nullif(1.0, 2);
+----
+1
+
+query error DataFusion error: Error during planning: Internal error: Failed to match any signature, errors: Error during planning: The signature expected NativeType::String but received NativeType::Int64
```

Comment:
This used to work, I just ran this locally against v43. I can't see a reason why this should no longer be supported.

Reply:
`select nullif(2, 'a');`

This query failed in Postgres, I think we should return error for this query